internal/controller/buildpipeline/buildpipeline_adapter.go

```diff
@@ -189,6 +195,69 @@ func (a *Adapter) EnsurePRGroupAnnotated() (controller.OperationResult, error) {
 	return controller.ContinueProcessing()
 }
 
+// EnsureIntegrationTestReportedToGitProvider is an operation that will ensure that the integration test status is initialized as pending
+// state when build PLR is triggered/retriggered or cancelled/failed when failing to create snapshot
+// to prevent PR/MR from being automerged unexpectedly and also show the snapshot creation failure on PR/MR
+func (a *Adapter) EnsureIntegrationTestReportedToGitProvider() (controller.OperationResult, error) {
+	if tekton.IsPLRCreatedByPACPushEvent(a.pipelineRun) {
+		a.logger.Info("build pipelineRun is not created by pull/merge request, no need to set integration test status in git provider")
+		return controller.ContinueProcessing()
+	}
+
+	if metadata.HasAnnotation(a.pipelineRun, tekton.SnapshotNameLabel) {
+		a.logger.Info("snapshot has been created for build pipelineRun, no need to report integration status from build pipelinerun status")
+		if err := tekton.AnnotateBuildPipelineRun(a.context, a.pipelineRun, h.SnapshotCreationReportAnnotation, "SnapshotCreated", a.client); err != nil {
```

Comment:
Should we first check if the Build pipelineRun is already annotated before attempting to annotate it?

Reply:
Actually since we're calling a `patch` command in [AnnotateBuildPipelineRun](https://github.com/konflux-ci/integration-service/blob/a0ef20f5bdade19fc1409ebd17cb04fed90aed1c/tekton/build_pipeline.go#L55), this shouldn't update the resourceVersion of the CR on the cluster so I think we should be fine.